File: Python/containsNearByDuplicate.py
def containsNearbyDuplicate(nums, k):
    """
    :type nums: List[int]
    :type k: int
    :rtype: bool
    """
    di = {}
    for i in xrange(len(nums)):
        print di
        if nums[i] in di:
            index = di.get(nums[i])
            print i,index
            if i- index <= k:
                return True
        di[nums[i]] = i
    return False
        
    
print containsNearbyDuplicate([1,2,1],0)
print containsNearbyDuplicate([1,2,1],1)
print containsNearbyDuplicate([-1,-1],1)
print containsNearbyDuplicate([1,0,1,1],1)

File: src/containsNearbyDuplicate.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 * [5,4,3,2,0,2] --> true
 * [1,0,1,1] --> true
 */
var containsNearbyDuplicate = function(nums, k) {
    var myMap = new Map();
    for(var i=0;i<nums.length;i++){
        var key = nums[i];
        if(myMap.has(key) && i - myMap.get(key) <= k){ // if key exists and satisfy k, return true
            return true;
        } else {
            myMap.set(key, i); // need to add new key-value pair or update the index of existing number
        }
    }
    // already iterated array and didn't find expected answer
    return false;
};